mm2src/mm2_main/tests/docker_tests_main.rs

```diff
@@ -100,6 +102,27 @@ pub fn docker_tests_runner(tests: &[&TestDescAndFn]) {
     test_main(&args, owned_tests, None);
 }
 
+fn wait_for_geth_node_ready() {
+    let mut attempts = 0;
+    loop {
+        if attempts >= 10 {
+            println!("Failed to connect to Geth node after several attempts.");
+            break;
+        }
+        match block_on(GETH_WEB3.eth().block_number()) {
```

Comment:
If the server responds very slowly, this can increase the amount of time we spend on tests significantly. We should add a timeout (~6 seconds for example) on the `GETH_WEB3.eth().block_number()` call.

Reply:
Hmm here we decided to move from 5 attempts and 5s duration time to not to wait too long :D 
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2100#discussion_r1596480544
I suggest to decrease attempts to 5 and set 3s sleep duration. This should be enough.

This functionality was added when I had nft swap tests on Geth node. I thought that node may have issues with syncing so I decided to wait for the connection to be established to definitely eliminate this problem. But as further tests showed that the issue wasnt in Geth connection. Well nft swaps have problems in geth but node receives all requests.

actually, if you think that `wait_for_geth_node_ready` is just holding the process and increasing tests time we can remove this function.

Reply:
Adding a timeout should be enough and simple, like this one: https://github.com/KomodoPlatform/komodo-defi-framework/blob/29c48bb3a57cde0976ce41d5f47ffed3fa110542/mm2src/coins/tendermint/tendermint_coin.rs#L302

Reply:
added timeout
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2100/commits/a2c47f8680d536038b2dd168f6500375efeb65b8